src/rpc_client/mod.rs

```diff
@@ -410,6 +410,16 @@ impl EthRpcClient {
         .reduce(self.consensus_strategy())
     }
 
+    pub async fn eth_estimate_gas(&self, params: EthCallParams) -> ReducedResult<Data> {
+        self.parallel_call(
+            "eth_estimateGas",
+            params,
+            self.response_size_estimate(512 + HEADER_SIZE_LIMIT),
```

Comment:
nit: I would a priori not expect that call to produce bigger responses that `eth_call`
```suggestion
            self.response_size_estimate(256 + HEADER_SIZE_LIMIT),
```